src/main/java/week1/db/SqlUserServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SqlUserServiceImpl implements SqlUserService {
+
+    public static final String URL = PathUtils.getDbPath();
+    public static final String USER = PathUtils.getDbLogin();
+    public static final String PASSWORD = PathUtils.getDbPass();
+    private Connection connection;
+
+    private PreparedStatement preparedStatement;
+    private Statement statement;
+
+    public SqlUserServiceImpl() throws SQLException {
+        Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+    }
+
+    @Override
+    public User create(String firstName, String lastName, int age, Address address) throws SQLException {
+
+        int id = findMaxUserId() + 1;
+
+        User user = new User(id, firstName, lastName, age, address);
+        preparedStatement = connection.prepareStatement
+                ("INSERT INTO users(first_name, last_name, age, adress_id) VALUES (?,?,?,?)");
+        preparedStatement.setString(1, user.getFirst_name());
+        preparedStatement.setString(2, user.getLast_name());
+        preparedStatement.setInt(3, user.getAge());
+        preparedStatement.setInt(4, setAddressId(address.getCity(), address.getStreet(), address.getHouse()));
+
+        preparedStatement.execute();
+
+        return user;
+    }
+
+    @Override
+    public List<User> findByFistNameAndLastName(String firstName, String lastName) throws SQLException {
+
+        String selectSQL = "SELECT * FROM users WHERE first_name = ? AND last_name = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(selectSQL);
+        preparedStatement.setString(1, firstName);
+        preparedStatement.setString(2, lastName);
+        ResultSet resultSet = preparedStatement.executeQuery(selectSQL);
+
+        return createUserListFromSet(resultSet);
+    }
+
+    @Override
+    public User changeAddress(Integer userId, Address newAddress) throws SQLException {
+
+        Address curAddress = getAddressById(userId);
+
+        if (newAddress.getId() == curAddress.getId()) {
+
+            return getUserById(userId);
+
+        } else if (addressPresentInDb(newAddress.getCity(), newAddress.getStreet(), newAddress.getHouse())) {
+
+            changeAddressId(userId, newAddress);
+
+            return getUserById(userId);
+
+        } else {
+            addNewAddressToDB(newAddress.getCity(), newAddress.getStreet(), newAddress.getHouse());
+            changeAddressId(userId, newAddress);
+
+            return getUserById(userId);
+        }
+    }
+
+    @Override
+    public void delete(Integer userId) throws SQLException {
+
+        preparedStatement = connection.prepareStatement
+                ("DELETE FROM users WHERE id = ?");
+
+        preparedStatement.setInt(1, userId);
+        preparedStatement.execute();
+
+    }
+
+    private void changeAddressId(Integer userId, Address newAddress) throws SQLException {
```

Comment:
this method does nothing. Most probably, you forgot to execute at the end.

Reply:
this method change address_id value in USERS table in db, taking it from address I tranmit in method

Reply:
thanks)